exec.go

```diff
@@ -124,20 +124,17 @@ func getSubCgroupPaths(args []string) (map[string]string, error) {
 	paths := make(map[string]string, len(args))
 	for _, c := range args {
 		// Split into controller:path.
-		cs := strings.SplitN(c, ":", 3)
-		if len(cs) > 2 {
-			return nil, fmt.Errorf("invalid --cgroup argument: %s", c)
-		}
-		if len(cs) == 1 { // no controller: prefix
+		if ctr, path, ok := strings.Cut(c, ":"); ok {
+			// There may be a few comma-separated controllers.
+			for _, ctrl := range strings.Split(ctr, ",") {
+				paths[ctrl] = path
+			}
+		} else {
+			// No controller: prefix.
 			if len(args) != 1 {
 				return nil, fmt.Errorf("invalid --cgroup argument: %s (missing <controller>: prefix)", c)
 			}
 			paths[""] = c
```

Comment:
Was trying to somewhat grasp what this case is for;

- if zero args are passed, we return early (start of function)
- if 1 arg is passed, it _MUST_ have a controller prefix
- But if 2 (or more) args are passed, then we don't have to?

So;

```bash
# valid: single arg
--cgroup a,b:/hello-world

# valid: single arg without controller
--cgroup /hello-world

# invalid: multiple args, one without controller
--cgroup a,b:/hello-world --cgroup /hello-world
--cgroup /hello-world --cgroup a,b:/hello-world 
```


So, if no controller is passed, it's for _everything_ which is only allowed if no other (per-controller) paths are specified, correct? (so not allowed to be combined).

☝️ Perhaps that's something we should capture in a comment on the function


Also;

- perhaps we need to change the `ok` check to also check if _controller_ is not empty, or is `,,,,,:/hello-world` something that should be considered valid?
- are duplicate controllers valid? They currently overwrite previous values (`foo,foo,foo:/bar`,  `foo:/baz`)